nuxeo-core-binarymanager-s3/src/main/java/org/nuxeo/ecm/core/storage/sql/S3DirectBatchHandler.java

```diff
@@ -143,18 +155,35 @@ public Batch getBatch(String batchId) {
 
         // create the batch
         Batch batch = new Batch(batchId, parameters, getName(), getTransientStore());
-        AssumeRoleRequest assumeRoleRequest = new AssumeRoleRequest().withRoleSessionName(batch.getKey())
-                                                                     .withRoleArn(roleArn);
-        Credentials credentials = stsClient.assumeRole(assumeRoleRequest).getCredentials();
+
+        String bucketPath = concatPath(bucketPrefix, batchId) + "/";
+
+        String policy = replaceStringWithPlaceholders(policyTemplate, REGEX_BUCKET_PATH_PLACE_HOLDER, bucketPath) + "*";
```

Comment:
No need to have `replaceStringWithPlaceholders`, it's one line and used only here. Would be easier to read if code is here instead of in the method.

Reply:
+1

Reply:
Why do you add the `*` here? Why not let the policy template specify it and avoid second-guessing the administrator?